src/systemTest/java/com/flightstats/hub/system/functional/WebhookLifecycleTest.java

```diff
@@ -0,0 +1,119 @@
+import com.flightstats.hub.kubernetes.HubLifecycle;
+import com.flightstats.hub.system.service.CallbackService;
+import com.flightstats.hub.system.service.ChannelService;
+import com.flightstats.hub.system.service.WebhookService;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import javax.inject.Inject;
+import java.util.Collections;
+import java.util.List;
+
+import static com.flightstats.hub.util.StringUtils.randomAlphaNumeric;
+import static junit.framework.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@Slf4j
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class WebhookLifecycleTest extends DependencyInjector {
+    @Inject
+    private CallbackService callbackResource;
+    @Inject
+    private ChannelService channelResource;
+    @Inject
+    private WebhookService webhookResource;
+    @Inject
+    private HubLifecycle hubLifecycle;
+    @Inject
+    private ModelBuilder modelBuilder;
+
+    private String channelName;
+    private String webhookName;
+
+    @BeforeAll
+    void hubSetup() {
+        hubLifecycle.setup();
+    }
+
+    @BeforeEach
+    void before() {
+        channelName = randomAlphaNumeric(10);
+        webhookName = randomAlphaNumeric(10);
+    }
+
+    private Webhook buildWebhook() {
+        return modelBuilder.webhookBuilder()
+                .channelName(channelName)
+                .webhookName(webhookName)
+                .build();
+    }
+
+    @Test
+    @SneakyThrows
+    void testWebhookWithNoStartItem() {
+        String data = "{\"fn\": \"first\", \"ln\":\"last\"}";
+
+        channelResource.create(channelName);
+
+        Webhook webhook = buildWebhook().withParallelCalls(2);
+        webhookResource.insertAndVerify(webhook);
+
+        List<String> channelItems = channelResource.addItems(channelName, data, 10);
+        assertTrue(callbackResource.areItemsEventuallySentToWebhook(webhookName, channelItems));
+    }
+
+    @Test
+    @SneakyThrows
+    void testWebhookWithStartItem() {
+        String data = "{\"key1\": \"value1\", \"key2\":\"value2\"}";
+
+        channelResource.create(channelName);
+        List<String> channelItems = channelResource.addItems(channelName, data, 10);
+
+        Webhook webhook = buildWebhook().
+                withStartItem(channelItems.get(4)).
+                withParallelCalls(2);
+        webhookResource.insertAndVerify(webhook);
+        List<String> channelItemsExpected = channelItems.subList(5, channelItems.size());
+        assertTrue(callbackResource.areItemsEventuallySentToWebhook(webhookName, channelItemsExpected));
+    }
+
+    @Test
+    @SneakyThrows
+    void testWebhookWithStartItem_expectItemsInOrder() {
```

Comment:
based on the name of this test I expect it to substantially different from the previous one, as far as I can tell it is only the data itself that changed, and the parallelism, unless there is an additional assertion I think this might work as a parameterized test

Reply:
I'll take a look.

Reply:
I'll add an extra order assertion for the test and give it a clearer name. I'll save parameterization for another PR.